cmd/azure.go

```diff
@@ -0,0 +1,333 @@
+	if length == -1 {
+		byteRange = fmt.Sprintf("%d-", startOffset)
+	}
+	rc, err := a.client.GetBlobRange(bucket, object, byteRange, nil)
+	if err != nil {
+		return traceError(azureToObjectError(err, bucket, object))
+	}
+	io.Copy(writer, rc)
+	rc.Close()
+	return nil
+}
+
+// GetObjectInfo - Use Azure equivalent GetBlobProperties.
+func (a AzureObjects) GetObjectInfo(bucket, object string) (objInfo ObjectInfo, err error) {
+	prop, err := a.client.GetBlobProperties(bucket, object)
+	if err != nil {
+		return objInfo, traceError(azureToObjectError(err, bucket, object))
+	}
+	t, err := time.Parse(time.RFC1123, prop.LastModified)
+	if err != nil {
+		return objInfo, traceError(err)
+	}
+	objInfo.Bucket = bucket
+	objInfo.ContentEncoding = prop.ContentEncoding
+	objInfo.ContentType = prop.ContentType
+	objInfo.MD5Sum = prop.Etag
+	objInfo.ModTime = t
+	objInfo.Name = object
+	objInfo.Size = prop.ContentLength
+	return
+}
+
+// PutObject - Use Azure equivalent CreateBlockBlobFromReader.
+func (a AzureObjects) PutObject(bucket, object string, size int64, data io.Reader, metadata map[string]string, sha256sum string) (objInfo ObjectInfo, err error) {
+	err = a.client.CreateBlockBlobFromReader(bucket, object, uint64(size), data, nil)
+	if err != nil {
+		return objInfo, traceError(err)
+	}
+	return a.GetObjectInfo(bucket, object)
+}
+
+// CopyObject - Use Azure equivalent CopyBlob.
+func (a AzureObjects) CopyObject(srcBucket, srcObject, destBucket, destObject string, metadata map[string]string) (objInfo ObjectInfo, err error) {
+	err = a.client.CopyBlob(destBucket, destObject, a.client.GetBlobURL(srcBucket, srcObject))
+	if err != nil {
+		return objInfo, traceError(azureToObjectError(err, srcBucket, srcObject))
+	}
+	return a.GetObjectInfo(destBucket, destObject)
+}
+
+// DeleteObject - Use azure equivalent DeleteBlob.
+func (a AzureObjects) DeleteObject(bucket, object string) error {
+	err := a.client.DeleteBlob(bucket, object, nil)
+	if err != nil {
+		return traceError(azureToObjectError(err, bucket, object))
+	}
+
+	return nil
+}
+
+// ListMultipartUploads - Incomplete implementation, for now just return the prefix if it is an incomplete upload.
+func (a AzureObjects) ListMultipartUploads(bucket, prefix, keyMarker, uploadIDMarker, delimiter string, maxUploads int) (result ListMultipartsInfo, err error) {
+	result.MaxUploads = maxUploads
+	result.Prefix = prefix
+	result.Delimiter = delimiter
+	resp, err := a.ListObjectParts(bucket, prefix, prefix, 1, 1000)
+	if err != nil {
+		return result, nil
+	}
+	if len(resp.Parts) > 0 {
+		result.Uploads = []uploadMetadata{{prefix, prefix, time.Now().UTC(), ""}}
+	}
+	return result, nil
+}
+
+// NewMultipartUpload - Use Azure equivalent CreateBlockBlob.
+func (a AzureObjects) NewMultipartUpload(bucket, object string, metadata map[string]string) (uploadID string, err error) {
+	// Azure does not support multiple upload ids for an object. Hence we use object name itself as the uploadID.
```

Comment:
well explained, i will update the comment with the same content.

though we can use "azure" instead of ASB as in our context azure means azure-blob-storage and more easily readable.